### ISSUE TITLE
this shit needs tests

### ISSUE THREAD

**Issue by rowanlupton** [rowanlupton/pylodon#73]
and i haven't learnt how to write them

**Comment by bhaugen** [comment]
@rowanlupton are you still working on this?
Has anybody used it for a real live ActivityPub?
We've kicked the tires a bit because we know python but haven't figured out how to deploy a real AP with it yet.

**Comment by rowanlupton** [comment]
hey! i haven't worked on it in a while. where i've got it to is, i could get posts to show up on mastodon? i haven't implemented much beyond that

**Comment by rowanlupton** [comment]
i got real work, and that kind of put big side projects like this on hold. so did trying to figure out how manage authentication ;)

**Comment by bhaugen** [comment]
Uh-oh, that reall work will get ya!  But thanks for responding.

**Comment by mshayden** [comment]
The message signing stuff looks like the tricky part. Eugen's blog post looks like a good summary though...

https://blog.joinmastodon.org/2018/06/how-to-implement-a-basic-activitypub-server/

Maybe I will try to make time to work of AP stuff again. Yeah real work stuff does get in the way.

**Comment by rowanlupton** [comment]
yeah. i _think_ that i got signing to work, at least sometimes? posts wouldn't have shown up on mastodon at all otherwise, right? [in here](https://github.com/rowanlupton/pylodon/blob/f7cfa6745952743a110388f721670d1872ed7e6d/pylodon/utilities.py#L68)
